src/lambda-edge/shared/shared.ts

```diff
@@ -47,27 +49,118 @@ interface ConfigFromDisk {
     mode: Mode,
     httpHeaders: HttpHeaders;
     clientSecret: string;
+    nonceSigningSecret: string;
+    logLevel: keyof typeof LogLevel;
+    secretAllowedCharacters?: string;
+    pkceLength?: number;
+    nonceLength?: number;
+    nonceMaxAge?: number;
 }
 
+enum LogLevel {
+    'none' = 0,
+    'info' = 10,
+    'warn' = 20,
+    'error' = 30,
+    'debug' = 40,
+}
+
+class Logger {
+    constructor(private logLevel: LogLevel) { }
+
+    private jsonify(args: any[]) {
+        return args.map((arg: any) => {
+            if (typeof arg === 'object') {
+                try {
+                    return JSON.stringify(arg);
+                } catch {
+                    return arg;
+                }
+            };
+            return arg;
+        });
+    }
+    public info(...args: any) {
+        if (this.logLevel >= LogLevel.info) {
+            console.log(...this.jsonify(args));
+        }
+    }
+    public warn(...args: any) {
+        if (this.logLevel >= LogLevel.warn) {
+            console.warn(...this.jsonify(args));
+        }
+    }
+    public error(...args: any) {
+        if (this.logLevel >= LogLevel.error) {
+            console.error(...this.jsonify(args));
+        }
+    }
+    public debug(...args: any) {
+        if (this.logLevel >= LogLevel.debug) {
+            console.trace(...this.jsonify(args));
+        }
+    }
+}
 
 export interface Config extends ConfigFromDisk {
     tokenIssuer: string;
     tokenJwksUri: string;
     cloudFrontHeaders: CloudFrontHeaders;
+    logger: Logger;
+    secretAllowedCharacters: string;
+    pkceLength: number;
+    nonceLength: number;
+    nonceMaxAge: number;
 }
 
+let CONFIG: Config | undefined = undefined;
+
+export function setConfig(config?: Config) {
+    if (!process.env.IS_TEST_MODE) {
+        throw new Error('This method is intended to be used in test mode only: env var IS_TEST_MODE needs to be set');
+    }
+    CONFIG = config;
+}
```

Comment:
I'm not sure, if need this specific function for testing purpose.
With jest.mock we can mock what we needed.
```
    const config = {
        clientId: 'clientId',
        oauthScopes: ['oauthScope1'],
        cognitoAuthDomain: 'cognitoAuthDomain',
        redirectPathSignIn: 'redirectPathSignIn',
        redirectPathAuthRefresh: 'redirectPathAuthRefresh',
        cookieSettings: {nonce: 'test_nonce'},
        mode: 'mode',
        ...
    };
    fs.readFileSync.mockImplementation(() => {
        return config;
    });
```

So, getConfig() should still get us config.

Reply:
Allright, I'll remove that